android/app/src/main/java/org/ZingoLabs/Zingo/MainActivity.kt

```diff
@@ -1,44 +1,124 @@
 package org.ZingoLabs.Zingo
 
-import android.content.Intent
+import android.os.Build
 import android.os.Bundle
 import android.util.Log
+import androidx.annotation.RequiresApi
+import androidx.work.*
 import com.facebook.react.ReactActivity
-import java.util.concurrent.TimeUnit
+import kotlin.random.Random
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.days
+import kotlin.time.Duration.Companion.hours
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+import kotlin.time.toJavaDuration
+import kotlinx.datetime.Clock
+import kotlinx.datetime.DateTimeUnit
+import kotlinx.datetime.Instant
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.atTime
+import kotlinx.datetime.toInstant
+import kotlinx.datetime.toLocalDateTime
+import kotlinx.datetime.until
 
 class MainActivity : ReactActivity() {
     /**
      * Returns the name of the main component registered from JavaScript. This is used to schedule
      * rendering of the component.
      */
-    override fun getMainComponentName(): String? {
+    private val taskID = "Zingo_Processing_Task_ID"
+    private val SYNC_PERIOD = 24.hours
+    private val SYNC_DAY_SHIFT = 1.days // Move to tomorrow
+    private val SYNC_START_TIME_HOURS = 3.hours // Start around 3 a.m. at night
+    private val SYNC_START_TIME_MINUTES = 60.minutes // Randomize with minutes until 4 a.m.
+    override fun getMainComponentName(): String {
         return "Zingo!"
     }
     override fun onCreate(savedInstanceState: Bundle?) {
-        Log.w("", "Starting main activity")
-        val service = Intent(applicationContext, BackgroundSync::class.java)
-        applicationContext.stopService(service)
+        Log.w("ON_CREATE", "Starting main activity")
+        // cancel the task if it is in execution now
+        cancelExecutingTask()
```

Comment:
Why do we need to close it here and then reschedule? Is it about the clashing with the block sync run from the foreground?

Reply:
Actually Yes, The App in foreground start a different sync process in the react-native layer... And we don't want to have two active processes.